### DISCUSSION TITLE
How Does Doxygen Create File Names for HTML Output?

### DISCUSSION THREAD

**Discussion by reifsnid** [doxygen/doxygen#11191] [Q&A] [answered]
I've been wondering how Doxygen generates the hash that's used in output HTML file names and whether this hash is certain to be preserved across multiple runs? Specifically, this is when `SHORT_NAMES` is set to `NO`.

I'm also wondering if moving symbols in a header file will change the hash, or if a new overload function added in between existing overloads will alter any of the existing overloads' hash?

Essentially, are there any concerns I should that the hash/file name for symbols will change?

Thank you!

Daniel

**Comment by albert-github** [comment, marked as answer]
- The hash should be persistent over different runs (and different doxygen versions). 
- moving items in a header file doesn't change the hash, the hash is based on the, input, filename not the content
